packages/wallet-frontend/src/pages/transfer/pay.tsx

```diff
@@ -13,26 +13,70 @@ import { accountService } from '@/lib/api/account'
 import { paySchema, transfersService } from '@/lib/api/transfers'
 import { useDialog } from '@/lib/hooks/useDialog'
 import { SuccessDialog } from '@/components/dialogs/SuccessDialog'
-import { getObjectKeys } from '@/utils/helpers'
-import { useState } from 'react'
+import { getObjectKeys, transformAmount } from '@/utils/helpers'
+import { useEffect, useState } from 'react'
 import { paymentPointerService } from '@/lib/api/paymentPointer'
 import { ErrorDialog } from '@/components/dialogs/ErrorDialog'
+import { useDebounce } from '@/lib/hooks/useDebounce'
+import { assetService } from '@/lib/api/asset'
 
 type PayProps = InferGetServerSidePropsType<typeof getServerSideProps>
 
 export default function Pay({ accounts }: PayProps) {
+  let scale = 2
```

Comment:
Wouldn't it be better to return the transformed amount directly from backend? This way we would get rid of some logic in the client side.